Cargo.toml

```diff
@@ -2,7 +2,9 @@
 members = [
   "flake8_to_ruff",
   "ruff_dev",
+  "ruff_cli",
 ]
+default-members = [".", "ruff_cli"]
```

Comment:
I have some thoughts, but in your opinion, how does this compare to the following alternative?

> Move everything in `src` into `ruff_core`, keep `ruff_cli` as it is (roughly), then change the root to point to `ruff_cli/src/main.rs` or appropriate.

This is, IIUC, how [`ripgrep`](https://github.com/BurntSushi/ripgrep) is structured.

Right now, there's a slight oddity here in that the top-level project isn't what's being packaged and distributed (which relates to having to change the working directory on CI).

Reply:
(It would also be ok to say "We want to do this later, since it's another big change, but it is planned." I'm mostly trying to understand and align on the desired end-state.)

Reply:
I consider the "core" of ruff to be conceptually separate from the rule implementations, even if they are currently entangled. So I think the crate of the currently top-level `src` directory should keep the name `ruff`, as long as it contains both the core and the rule implementations.

While I think it makes some sense that the top-level `src` directory contains the code that contributors are most likely interested in editing, I'd be alright with moving `src` to `ruff/src` for the sake of consistency, however we might firstly want to think about further splitting up the crate, see #1820.